tools/tests/on-disk-format-checker.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
```

Comment:
lgtm. 


Some nits:

Calling cleanup using an EXIT handler may be simpler and save a couple of calls to _cleanup.

Example:
```
trap cleanup EXIT
function cleanup()
{

}
```

Consider removing the "_" function prefix. It is a pythonism for "private" fuctions, not? In this small script it IMHO doesn't add much value to the reader.


Consider set -x instead of the debug print logic.

Using set -e and if command or if ! command could be clearer for the reader to understand what may or may not fail.

Reply:
I am not a fan of `set -x` most of the time, especially not as default since it makes the output hard to read.
The exit handler is a good call though.

The thing with the `_` prefixes is something I like to do to a) signify that should someone source this file these functions aren't meant to be used willy-nilly and b) to avoid naming conflicts with stuff on `$PATH`. I also makes it quite clear that these aren't some command from somewhere but functions in this script. So in a sense these _are_ private functions. But I'll remove them if you think this is detrimental to readability.